youtube_dl/extractor/neteasemusic.py

```diff
@@ -140,7 +140,7 @@ def query_api(self, endpoint, video_id, note):
 class NetEaseMusicIE(NetEaseMusicBaseIE):
     IE_NAME = 'netease:song'
     IE_DESC = '网易云音乐'
-    _VALID_URL = r'https?://music\.163\.com/(#/)?song\?id=(?P<id>[0-9]+)'
+    _VALID_URL = r'https?://(y\.)?music\.163\.com/(#/|m/)?song\?.*?\bid=(?P<id>[0-9]+)'
```

Comment:
Make pattern group anonymous since not referenced:
```suggestion
    _VALID_URL = r'https?://(y\.)?music\.163\.com/(?:[#m]/)?song\?.*?\bid=(?P<id>[0-9]+)'
```